met-web/src/routes/LanguageParam.tsx

```diff
@@ -0,0 +1,45 @@
+import React, { useEffect, ComponentType } from 'react';
+import { useParams, useNavigate } from 'react-router-dom';
+import { useAppSelector } from '../hooks';
+import { LanguageState } from 'reduxSlices/languageSlice';
+
+interface RouteParams {
+    [key: string]: string | undefined;
+    engagementId?: string;
+    slug?: string;
+    subscriptionStatus?: string;
+    scriptionKey?: string;
+    dashboardType?: string;
+    token?: string;
+    widgetId?: string;
+}
+
+const withLanguageParam = <P extends object>(Component: ComponentType<P>) => {
+    return (props: P) => {
+        const languageState: LanguageState = useAppSelector((state) => state.language);
+        const languageCode = languageState.id;
+        const params = useParams<RouteParams>();
```

Comment:
Do we need to memoize the params object ? may help prevent unnecessary re-renders.